### PR TITLE
Added support for registering on process exec events

### PR DESCRIPTION
**Pull request by IlayRosenberg**
Also including a fix to the listener loop which would previously crash when handled too many events, and ignoring missing imports for the psutil library which failed `lint.sh` for me

### CODE REVIEW

File: granulate_utils/linux/proc_events.py [hunk 1]
```diff
@@ -116,7 +124,14 @@ def _listener_loop(self):
...
 
             for key, _ in events:
-                data = key.fileobj.recv(256)
+                try:
+                    # When stressed, reading from the socket can raise
+                    #   OSError: [Errno 105] No buffer space available
+                    # This seems to be safe to ignore, empirically no events were missed
+                    data = key.fileobj.recv(256)
+                except OSError as e:
+                    if e.errno == 105:
```
Comment (YishaiZinkin) [inline granulate_utils/linux/proc_events.py:133]: Do you suppress other exceptions on purpose?

Comment (IlayRosenberg) [reply granulate_utils/linux/proc_events.py:133]: woops

File: granulate_utils/linux/proc_events.py [hunk 2]
```diff
@@ -116,7 +124,14 @@ def _listener_loop(self):
                 break
 
             for key, _ in events:
-                data = key.fileobj.recv(256)
+                try:
+                    # When stressed, reading from the socket can raise
+                    #   OSError: [Errno 105] No buffer space available
+                    # This seems to be safe to ignore, empirically no events were missed
```
Comment (YishaiZinkin) [inline granulate_utils/linux/proc_events.py:130]: Interesting. Did you see it when there was a large amount of processes or on general network stress?

Comment (IlayRosenberg) [reply granulate_utils/linux/proc_events.py:130]: Large amount of processes. I stressed it on purpose locally

Comment (YishaiZinkin) [reply granulate_utils/linux/proc_events.py:130]: Hmm maybe I'll add packet filtering at some point which might solve this

File: mypy.ini [hunk 3]
```diff
@@ -2,3 +2,6 @@
 python_version = 3.6
 warn_unused_configs = True
 exclude = venv/bin/
+
+[mypy-psutil.*]
+ignore_missing_imports = True
```
Comment (YishaiZinkin) [inline mypy.ini:7]: It works for me, it seems like a local problem on your side. Maybe `types-psutil` isn't installed correctly? It appears in `dev-requirements.txt`.
Try to remove this and run the CI again. If it passes I don't think we should add these lines.

**Review by YishaiZinkin — APPROVED** [review]
Cool

**Pull request merged**